### ISSUE TITLE
Make sid and rid configurable

### ISSUE THREAD

**Issue by mosch** [legastero/stanza#78]
To be able to [prebind](https://github.com/priyadi/roundcube-converse.js-xmpp-plugin/tree/master/php/xmpp-prebind-php) the connection, sid and rid must be configurable.

Would you welcome a PR that implements this? Or did i maybe miss something?

**Comment by legastero** [comment]
You should be able to do it with:

```javascript
client = XMPP.createClient({
     ...,
     rid: 'the-rid',
     sid: 'the-sid'
});
```

The main client config gets passed to the transport's `connect()` method, so the `rid`/`sid` are set there (see https://github.com/otalk/stanza.io/blob/master/lib/transports/bosh.js#L109).

**Comment by mosch** [comment]
thanks for the response, i'm going to try it :+1:

**Comment by JeanCarriere** [comment]
It does not seem to work.
BOSHConnection.prototype.connect does not pass the sid.
On the other hand, if sid is passed, when connect() is called, will it create the session with this sid ?

**Comment by wilwade** [comment]
I also tried this and can report that it does not work. The sid is not set and the http-bind request tries a different sid:

```
var client = XMPP.createClient({
    jid: 'person@example.org/resource',
    sid: '6af79e30-d8fd-4417-83ff-142cb34db8a1',
    rid: 6287756265,
    transport: 'bosh',
    boshURL: 'https://www.example.org/http-bind'
});
client.connect()
```

Logs:
```
Apr 09 15:10:48 mod_bosh	debug	BOSH body open (sid: <none>)
Apr 09 15:10:48 boshef0eb8b4-09da-4ad0-b787-1b6f6a90c647	debug	BOSH session created for request from 192.168.56.101
Apr 09 15:10:48 mod_bosh	info	New BOSH session, assigned it sid 'ef0eb8b4-09da-4ad0-b787-1b6f6a90c647'
```

**Comment by ethanfrogers** [comment]
After looking over this code for a project, it seems that the connect is what is needed to get the SID so if you already have an SID, technically all it needs to do is finish the setup. I have been working on figuring it out.

**Comment by jrouly** [comment]
Sorry to resurrect and old thread, but I was wondering if @ethanfrogers @wilwade or @mosch got their code to work by passing in RID/SID? I'm looking into attempting the same prebinding application using Stanza and the XMPP-Prebind example linked above.

**Comment by ethanfrogers** [comment]
@jrouly I am no longer working on that project, but this issue was one of the reasons we went a different direction. I hacked something together by bypassing some of the initial setup that happens during the connection process, but it didn't work as well as I would have liked. Unfortunately, I don't have any of that code anymore. It is possible and I was able to successfully send messages using an already provided SID, however.
